src/west/app/project.py

```diff
@@ -202,63 +202,65 @@ def do_run(self, args, ignored):
         else:
             topdir = self.bootstrap(args)
 
-        log.banner(f'Initialized. Now run "west update" inside {topdir}.')
+        log.banner('Initialized. Now run "west update" inside',
+                   topdir)
```

Comment:
just curios, here you change into two args, but in line 245, you do the opposite:
```
log.banner(f'Initializing in {topdir}')
```
any particular reason ?

Reply:
It was when I started changing this file that I began learning more about the differences between `f'{foo}'`, `str(foo)`, `repr(foo)`, and `os.fspath(foo)` when `foo` is either a general-purpose `os.PathLike` or specifically a `pathlib.Path`. I didn't know `os.fspath` existed when I started this work and so I was relying on behavior that is specific to `pathlib.Path` in some cases, which I later learned was incorrect. Live and learn... and then you read the docs.

Any inconsistencies here are due to me not cleaning up after learning more. I will double check and make sure everything is OK. Thanks for asking.

Reply:
>  I will double check and make sure everything is OK.

I didn't notice any bugs being introduced, but I went back and removed the useless changes to print / log lines to not clutter the diff with unrelated changes.

For reference:

- `f'{foo}'` uses the `__format__` method from `foo`
- pathlib objects do not override `__format__`, so `f'{pathobj}'` ends up calling `object.__format__` on `pathobj` with an empty `format_spec`
- [`object.__format__`](https://docs.python.org/3/reference/datamodel.html#object.__format__) with an empty `format_spec`  ends up calling `str()` on the object being formatted (with some differences between python 3.6 and later versions that don't matter to us)
- for pathlib objects, `os.fspath(pathobj)` and `str(pathobj)` are the same thing, because `__fspath__` delegates to `str()`
- `west.log` ends up calling `print()` on its arguments, which will print them out just like `str()` would format them

So TL;DR `west.log.inf(f'blah blah f{pathobj}')` and `west.log.inf('blah blah ', pathobj)` produce the same results, because `__format__` for these objects works like `str()`, which works like `os.fspath()`, so there's no differences in printing them with f-strings or as separate args, they always produce the right filesystem representation as a string, and we might as well not introduce pointless churn.

Where this *does* matter is in places like west.manifest, which accept `os.PathLike`. In the general case, `str(some_os_PathLike)` and `os.fspath(some_os_PathLike)` are *not* the same, because the first uses `__str__` and the second uses `__fspath__`, which don't need to be the same (though they are for pathlib).

That's why I tried to use os.fspath in places in project.py where we were not obviously just printing things out.

Reply:
> Any inconsistencies here are due to me not cleaning up after learning more.

I think you missed my point.
When looking at the diff: `index b0ed0d5..5c68170 100644`
then I noticed:
```
-        log.banner(f'Initialized. Now run "west update" inside {topdir}.')
+        log.banner('Initialized. Now run "west update" inside',
+                   topdir)
```
and then later:
```
-        log.banner('Initializing in', topdir)
+        log.banner(f'Initializing in {topdir}')
```
INSIDE the same commit

No bugs, just curios why one place was changed from `(f'text {topdir}')`--> `('text', topdir)`, and at the second location you went from: `('text', topdir)` --> `(f'text {topdir}')`

Reply:
> I think you missed my point.

I don't think I did. I went back and made those consistent by dropping the unrelated changes to format strings... did I miss one? Anyway, not important.